rfc/0000-configuration-and-shared-files.md

```diff
@@ -0,0 +1,264 @@
+
+Configuration files complement the use of environment variables. They can allow for more complex configuration data to be stored, and can be used to store default values shared across the Pact ecosystem without requiring the end user to set environment variables all the time.
+
+This RFC also proposes to have both local (or per-project) configuration files, and user configuration files. This allows a user of Pact to configure their environment in a way that is consistent across all projects, as well as allowing for project-specific configuration.
+
+For example:
+
+- A user may wish to set the their default Pact Broker to `https://pact-broker.local.company.com` in their user configuration file so that all projects use this value by default.
+- A specific project might still be in development, and use a local Pact Broker at `http://localhost:9292` for testing. This could be set in the project's local configuration file.
+- The CI/CD pipeline still configures the Pact Broker using environment variables, as it is not practical to store configuration files in the pipeline.
+
+#### Precedence
+
+The following order of precedence _must_ be followed when reading configuration data:
+
+1. Environment variables
+2. Configuration files, in the following order:
+   1. If an explicit configuration file is set, only that file should be read.
+   2. If no explicit configuration file is set, the following files should be read in order:
+      1. Local configuration file
+      2. User configuration file
+3. Default values
+
+The explicit configuration file may either be set by the `PACT_CONFIG_FILE` environment variable, or by a command line option.
+
+#### Location
+
+To ensure a consistent experience across the Pact ecosystem, the following configuration files are proposed to be used across the Pact ecosystem:
+
+- **User Configuration File**
+
+  Location:
+
+  - Unix and Unix-like systems:
+    - `$XDG_CONFIG_HOME/pact/config.toml` (which defaults to `~/.config/pact/config.toml` if `XDG_CONFIG_HOME` is not set)
+  - Windows:
+    - `%APPDATA%\pact\config.toml`
+  - Fallback:
+    - `~/.pact/config.toml`
+
+  > [!NOTE]
+  >
+  > macOS is considered a Unix-like system for the purposes of this RFC. Should an end-user prefer to use a different location (such as `~/Library/Preferences`), they may do so by setting the `XDG_CONFIG_HOME` environment variable.
+
+  If the operating system cannot be detected, the fallback location _should_ be used.
+
+  If the operating system is detected and the configuration file is found in the fallback location, a warning _should_ be displayed informing the user to relocate the configuration.
```

Comment:
should a warning appear, if the configuration file is found in the fallback location, if the user explicitly selected that location, for `PACT_CONFIG_FILE`.

Same q for 

>- `PACT_DATA_HOME`
>- Fallback: `~/.pact/`

> If the operating system is detected and the fallback location is found, a warning _should_ be displayed to inform the user to relocate the directory.


If I am on a supported system, and I set the values, to the same as the fallbacks, `~/.pact/config.toml` & `~/.pact`, should the warning be suppressed?

Reply:
This is indeed ambiguous. The short answer is no, I don't think a warning should not be raised if the configuration (either set through environment variables, or from parsing the config file) points to a fallback location.

Reply:
thanks dude